[DOC] Doc conversion and update for examples

outoftardis opened the pull request:
# Description

- Converted the remaining algorithms into RST:

  - [MSE](https://outoftardis.github.io/daal/dev_guide/algorithms/optimization-solvers/objective-functions/mse.html)
  - [Log Loss](https://outoftardis.github.io/daal/dev_guide/algorithms/optimization-solvers/objective-functions/logistic-loss.html)
  - [Cross Entropy](https://outoftardis.github.io/daal/dev_guide/algorithms/optimization-solvers/objective-functions/cross-entropy.html)
  - [SAGA](https://outoftardis.github.io/daal/dev_guide/algorithms/optimization-solvers/solvers/stochastic-average-gradient-accelerated-method.html)
  - [CDA](https://outoftardis.github.io/daal/dev_guide/algorithms/optimization-solvers/solvers/coordinate-descent.html)
  - [online](https://outoftardis.github.io/daal/dev_guide/algorithms/pca/computation-online.html) and [distributed](https://outoftardis.github.io/daal/dev_guide/algorithms/pca/computation-distributed.html) PCA: 
  - [AdaBoost MultiClass](https://outoftardis.github.io/daal/dev_guide/algorithms/boosting/adaboost-multiclass.html)
  - distributed Implicit [ALS](https://outoftardis.github.io/daal/dev_guide/algorithms/implicit_als/implicit-alternating-least-squares.html): 
    - [initialization](https://outoftardis.github.io/daal/dev_guide/algorithms/implicit_als/initialization-distributed.html)
    - [computation: training](https://outoftardis.github.io/daal/dev_guide/algorithms/implicit_als/computation-distributed-training.html#implicit-als-distributed-training)
    - [computation: prediction](https://outoftardis.github.io/daal/dev_guide/algorithms/implicit_als/computation-distributed-prediction.html#implicit-als-distributed-prediction)

- Fixed broken links in docs
- Updated the script that creates rst files for examples. Removed the links to the cpp_sycl examples, added oneAPI examples for DPC++ and C++.
- Added links to daal4py examples
- Added a note about Java and GPU

[Output](https://outoftardis.github.io/daal/)

napetrov commented:
/intelci: run

averbukh commented:
I see a lot of not rendered tags in the rich view mode. It's wrong rst tags or github rst render issue. If it's latter, do you have rendered content for that?

outoftardis commented:
> I see a lot of not rendered tags in the rich view mode. It's wrong rst tags or github rst render issue. If it's latter, do you have rendered content for that?

See the artifact for this PR: https://dev.azure.com/daal/DAAL/_build/results?buildId=3494&view=artifacts&type=publishedArtifacts

outoftardis commented:
@averbukh I've updated gh-pages in my fork with the output for this PR.

To view the rendered output for the converted topics, see:

- [MSE](https://outoftardis.github.io/daal/dev_guide/algorithms/optimization-solvers/objective-functions/mse.html)
- [Log Loss](https://outoftardis.github.io/daal/dev_guide/algorithms/optimization-solvers/objective-functions/logistic-loss.html)
- [Cross Entropy](https://outoftardis.github.io/daal/dev_guide/algorithms/optimization-solvers/objective-functions/cross-entropy.html)
- [SAGA](https://outoftardis.github.io/daal/dev_guide/algorithms/optimization-solvers/solvers/stochastic-average-gradient-accelerated-method.html)
- [CDA](https://outoftardis.github.io/daal/dev_guide/algorithms/optimization-solvers/solvers/coordinate-descent.html)
- [online](https://outoftardis.github.io/daal/dev_guide/algorithms/pca/computation-online.html) and [distributed](https://outoftardis.github.io/daal/dev_guide/algorithms/pca/computation-distributed.html) PCA: 
- [AdaBoost MultiClass](https://outoftardis.github.io/daal/dev_guide/algorithms/boosting/adaboost-multiclass.html)
- distributed Implicit [ALS](https://outoftardis.github.io/daal/dev_guide/algorithms/implicit_als/implicit-alternating-least-squares.html): 
  - [initialization](https://outoftardis.github.io/daal/dev_guide/algorithms/implicit_als/initialization-distributed.html)
  - [computation: training](https://outoftardis.github.io/daal/dev_guide/algorithms/implicit_als/computation-distributed-training.html#implicit-als-distributed-training)
  - [computation: prediction](https://outoftardis.github.io/daal/dev_guide/algorithms/implicit_als/computation-distributed-prediction.html#implicit-als-distributed-prediction)

napetrov commented:
Examples update looks good

PetrovKP commented on docs/source/dev_guide/algorithms/svm/support-vector-machine-classifier.rst:
```diff
@@ -230,36 +232,49 @@ Examples
     Batch Processing:
     
-    -  :ref:`svm_two_class_thunder_dense_batch.cpp`
+    - :ref:`cpp_svm_two_class_smo_dense_batch.cpp`
```
oneAPI C++ have 2 examples for SVM (SMO & thunder method)
```
:ref:`cpp_svm_two_class_thunder_dense_batch.cpp
:ref:`cpp_svm_two_class_smo_dense_batch.cpp
```

outoftardis replied:
fixed

PetrovKP commented on docs/source/dev_guide/algorithms/logistic_regression/logistic-regression.rst:
```diff
@@ -261,27 +263,29 @@ Examples
+    - :daal4py_sycl_example:`log_reg_binary_dense_batch.py`
+
   .. tab:: Python*
 
```
For python example missed log_reg_binary_dense_batch.py. See 294 line in this file

outoftardis replied:
fixed

PetrovKP commented:
I am confused by the different style of tables with parameters. In some places they are not stand out, in others they stand out. Is this expected?

For example:
Objective Function:
![image](https://user-images.githubusercontent.com/17429885/89119523-bb051e80-d4b7-11ea-8507-fbbe2dbf10f7.png)

The sum of functions algorithm:
![image](https://user-images.githubusercontent.com/17429885/89119532-c8220d80-d4b7-11ea-9417-a6755369d894.png)

outoftardis commented:
> I am confused by the different style of tables with parameters. In some places they are not stand out, in others they stand out. Is this expected?
> 
> For example:
> Objective Function:
> ![image](https://user-images.githubusercontent.com/17429885/89119523-bb051e80-d4b7-11ea-8507-fbbe2dbf10f7.png)
> 
> The sum of functions algorithm:
> ![image](https://user-images.githubusercontent.com/17429885/89119532-c8220d80-d4b7-11ea-9417-a6755369d894.png)

This is expected, yes. I'm fixing it gradually.

Vika-F commented on docs/source/dev_guide/algorithms/implicit_als/computation-distributed-training.rst:
```diff
@@ -0,0 +1,246 @@
+Algorithm Parameters
+********************
+
+At the training stage, implicit ALS recommender in the distributed processing mode has the following parameters:
+
```
It seems the table with parameters is missing here.

outoftardis replied:
fixed, thanks

Vika-F commented on docs/source/dev_guide/algorithms/implicit_als/computation-distributed-training.rst:
```diff
@@ -0,0 +1,246 @@
+.. image:: images/implicit-als-distributed-computation-training-step-3.png
+    :width: 600
+    :align: center
+
```
The text "In this step, implicit ALS recommender training accepts the input described below. ... " is missing here.

outoftardis replied:
fixed, thanks

Vika-F commented on docs/source/dev_guide/algorithms/pca/computation-batch.rst:
```diff
@@ -0,0 +1,175 @@
+   * - correlation
+     - Use when the input data is a correlation matrix. Pointer to the :math:`p \times p`
+       numeric table that contains the correlation matrix. This input can be an
+       object of any class derived from NumericTable except
```
In ALS algorithm the text "This input can be an object of any class ..." is formatted as a Note statement. Should it be formatted the same way across all algorithms?

outoftardis replied:
fixed

Vika-F commented on docs/source/dev_guide/algorithms/pca/computation-batch.rst:
```diff
@@ -0,0 +1,175 @@
+Algorithm Input
+---------------
+The PCA algorithm accepts the input described below. Pass the
+Input ID as a parameter to the methods that provide input for your algorithm.
```
"For more details, see Algorithms." is missing here.

outoftardis replied:
fixed

Vika-F commented on docs/source/dev_guide/algorithms/pca/computation-batch.rst:
```diff
@@ -0,0 +1,175 @@
+       in the descending order. By default, this result is an object of the
+       HomogenNumericTable class, but you can define the result as an object of
+       any class derived from NumericTable except PackedSymmetricMatrix,
+       PackedTriangularMatrix, and CSRNumericTable.
```
In ALS algorithm the text "This input can be an object of any class ..." is formatted as a Note statement. Should it be formatted the same way across all algorithms?
The same for 'eigenvectors'.

outoftardis replied:
Yes, you are right. But at the moment there are still instances where this text isn't converted into a note, this is not the only one. I'm changing this gradually.

outoftardis commented:
@PetrovKP @Vika-F I've addressed all of your comments.
There are still some inconsistencies in formatting in the doc, please ignore it for now. I'm aware that they exist and I'm working on fixing them.

outoftardis commented:
@averbukh @Vika-F @PetrovKP @napetrov do you have any other comments regarding this PR?

napetrov commented:
/intelci: run

napetrov commented:
@outoftardis  could you please rebase so internal ci can validate content.
Overall i don't have other comments

napetrov commented:
Looks good for me. @averbukh  - is there any questions/comments or we can go and merge?
